layers/cip.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2017 Google, Inc. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file in the root of the source
+// tree.
+
+package layers
+
+import (
+	"encoding/binary"
+	"errors"
+
+	"github.com/google/gopacket"
+)
+
+const (
+	cipBasePacketLen int = 2
+)
+
+var (
+	ErrCIPDataTooSmall = errors.New("CIP packet data truncated")
+)
+
+type CIP struct {
+	BaseLayer
+	Response         bool     // false if request, true if response
+	ServiceID        byte     // The service specified for the request
+	ClassID          uint16   // request only
+	InstanceID       uint16   // request only
+	Status           byte     // Response only
+	AdditionalStatus []uint16 // Response only
+	Data             []byte   // Command data for request, reply data for response
```

Comment:
Would it be better to make this a Payload layer within the CIP layer?

Reply:
I considered that but 1. the field is called Data in the spec and 2. As far as I know, there's no further layers on top of CIP. I'm willing to change it, but I thought I'd explain my reasoning first.